app/actions/views/channel.js

```diff
@@ -718,22 +721,42 @@ export function loadChannelsForTeam(teamId, skipDispatch = false, isReconnect =
             for (let i = 0; i <= MAX_RETRIES; i++) {
                 try {
                     console.log('Fetching channels attempt', (i + 1), teamId, 'include deleted since', lastConnectAt); //eslint-disable-line no-console
-                    const [channels, channelMembers] = await Promise.all([ //eslint-disable-line no-await-in-loop
+                    const [channels, channelMembers, categories] = await Promise.all([ //eslint-disable-line no-await-in-loop
                         Client4.getMyChannels(teamId, true, lastConnectAt),
                         Client4.getMyChannelMembers(teamId),
+                        Client4.getChannelCategories(currentUserId, teamId),
```

Comment:
I ended up moving getting the getChannelCategories as part of getMyChannels - I couldn't figure out a good way of getting the config.showLegacySidebar here; but it's only one extra http call - hardly adds any overhead.